app/controllers/admin/users_controller.rb

```diff
@@ -7,15 +7,45 @@ def show
     @user = User.find(params[:id])
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      flash[:success] = "User profile has been updated"
+      redirect_to admin_user_path(@user)
+    else
+      errors = @user.errors.details
+      if errors.has_key?(:email) && errors[:email].first[:error] == :taken
+        flash[:danger] = "That email is already registered."
+        @user.email = nil
+        render :'users/edit'
+      end
+    end
+
+  end
+
+
   def enable
     User.find(params[:id]).update(disabled: false)
-    flash[:primary] = 'You have enabled a user' if params[:disabled] == "false"
+    flash[:primary] = 'You have enabled a user'
     redirect_to admin_users_path
   end
 
   def disable
     User.find(params[:id]).update(disabled: true)
-    flash[:primary] = 'You have disabled a user' if params[:disabled] == "true"
```

Comment:
This was throwing me an error, the flash message wasn't appearing. This made it work

Reply:
Makes sense, with the new path it shouldn’t have had the conditional on the parameters.